CMakeLists.txt

```diff
@@ -344,8 +344,9 @@ if(MSVC)
 
 endif(MSVC)
 
-# C++11 support
-set(CMAKE_CXX_STANDARD 11)
+# C++17 support
+#set(CMAKE_CXX_STANDARD 17) // CMAKE version >= 3.8
+set(CMAKE_CXX_FLAGS "-std=c++17")
```

Comment:
You shouldn't switch back to the cmake_cxx_flag as some compilers need you to specify that differently (like msvc needs /std:c++ and I'm pretty sure gcc and clang are different.

Reply:
Agreed. I was reluctant to force a newer CMAKE version (I'm developing on a debian stable sytsem with comes with CMAKE 3.7.2). Fixed it.

Reply:
Fixed this by making the code C++11 compliant using your comments.